### PR TITLE
DP4-1 [Users] Access for different user roles

### PR DESCRIPTION
**Pull request by koleshkos**
I have two user roles: User and Admin. In order to differentiate access to the functionality, i created a separate URL for the Admin. 

Setup the following URLs for admin and user: 
URL for admin:
    [website_home_page]/admin - admin's homepage 
    [website_home_page]/admin/[page_name] - other pages
URL for user:
    [website_home_page]- user's hamepage 
    [website_home_page]/[page_name] - other pages

Feature is covered with tests

### CODE REVIEW

File: app/helpers/pages_helper.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# frozen_string_literal: true
+
+module PagesHelper
+end
```
Comment (DmitryGolubko) [inline app/helpers/pages_helper.rb]: remove empty helpers

File: app/views/admin/index.html.erb [hunk 2]
```diff
@@ -0,0 +1 @@
+<h1>Welcome ADMIN to app MY SHOP</h1>
```
Comment (DmitryGolubko) [inline app/views/admin/index.html.erb]: let's use haml or slim as template engine, as it's more popular and compact

File: config/routes.rb [hunk 3]
```diff
@@ -1,3 +1,6 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  root 'pages#home'
+  
+  resources :admin
```
Comment (DmitryGolubko) [inline config/routes.rb]: are we going to do crud operations on admin resources? Let's make just an admin scope and `resource :products` for products crud

File: spec/controllers/admin_spec.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
+require 'rails_helper'
+
+RSpec.describe AdminController do
+  describe "GET admin#index" do 
+    subject { get :index }
+
+    it "renders the index template" do      
+      expect(subject).to render_template(:index)
+      expect(subject).to render_template('index')
+      expect(subject).to render_template('admin/index')
```
Comment (DmitryGolubko) [inline spec/controllers/admin_spec.rb]: these expectations are the same, leave just one of them

File: spec/controllers/pages_spec.rb [hunk 5]
```diff
@@ -0,0 +1,17 @@
+require 'rails_helper'
+
+RSpec.describe PagesController do
+  describe "GET #pageshome" do 
+    subject { get :home }
+
+    it "renders the home template" do      
+      expect(subject).to render_template(:home)
+      expect(subject).to render_template('home')
+      expect(subject).to render_template('pages/home')
```
Comment (DmitryGolubko) [inline spec/controllers/pages_spec.rb]: the same

**Review by DmitryGolubko — CHANGES_REQUESTED** [review]
changes requested
TODO: 
1. Let's use the following git commits convention:
`Ticket number(DP4-1) - commit name`
2. Setup CircleCI for automatic tests and style checker running 
3. Setup Heroku app for automatic deploy when tests pass
Contact me or Danila if there are any questions

File: app/controllers/admin_controller.rb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+# frozen_string_literal: true
+
+class AdminController < ApplicationController
+  def index; end
+end
```
Comment (DmitryGolubko) [inline app/controllers/admin_controller.rb]: If you'll look at the following tasks, you'll see that you have to work with products resources, so you don't need admin controller here, as we don't have any admin model or smth like.

**Review by DmitryGolubko — CHANGES_REQUESTED** [review]
remove admin controller, and add rubocop check to circleCI

File: app/controllers/admin_controller.rb (hunk 6 above)
Comment (koleshkos) [reply app/controllers/admin_controller.rb]: I got it. I solve this problem

**Comment by koleshkos** [comment]
- Removed admin controller
- Added rubocop check to circleCI
- Setup CircleCi for test
- Used HAML
- Added specs

**Review by DmitryGolubko — APPROVED** [review]
Approved, but let's add Rubocop run in separate CircleCI job

**Comment by koleshkos** [comment]
Added Rubocop run in separate CircleCI job

**Review by DmitryGolubko — APPROVED** [review]
Approved

**Pull request merged**